backend/python/app/tests/graphql/mutations/test_story_mutation.py

```diff
@@ -0,0 +1,163 @@
+import re
+
+from ....models.story import Story
+
+
+def test_create_story(app, db, client):
+    result = client.execute(
+        """
+        mutation {
+            createStory(
+                contents: [
+                    "line 1", "line 2", "line 3",
+                ],
+                storyData: {
+                    title: "title test",
+                    description: "this should explain things",
+                    youtubeLink: "don't think we validate this yet",
+                    level: 10001
+                }
+            ) {
+                ok
+                story {
+                    id
+                    title
+                    description
+                    youtubeLink
+                    translatedLanguages
+                    level
+                    contents {
+                        id
+                        storyId
+                        lineIndex
+                        content
+                    }
+                }      
+            }
+        }
+    """
+    )
+
+    assert dict(result["data"]) == {
+        "createStory": {
+            "ok": True,
+            "story": {
+                "id": 1,
+                "title": "title test",
+                "description": "this should explain things",
+                "youtubeLink": "don't think we validate this yet",
+                "translatedLanguages": None,
+                "level": 10001,
+                "contents": [
+                    {"id": 1, "storyId": "1", "lineIndex": 0, "content": "line 1"},
+                    {"id": 2, "storyId": "1", "lineIndex": 1, "content": "line 2"},
+                    {"id": 3, "storyId": "1", "lineIndex": 2, "content": "line 3"},
+                ],
+            },
+        }
+    }
+
+    test_db_obj = (
+        Story.query.order_by(Story.id.desc())
+        .first()
+        .to_dict(include_relationships=True)
+    )
+    # will want this somewhere globally i assume
+    pattern = re.compile(r"(?<!^)(?=[A-Z])")
+
+    for field_name, value in result["data"]["createStory"]["story"].items():
+        if field_name == "contents":
+            value = [
+                {to_snake_case(pattern, k): v for k, v in content.items()}
+                for content in value
+            ]
+
+        assert value == test_db_obj[to_snake_case(pattern, field_name)]
+
+
+def test_create_story_exception(db, client):
+    pass
+
+
+def test_create_story_translation(db, client):
+    pass
+
+
+def test_create_story_translation_exception(db, client):
```

Comment:
@gaoxk 
> Not a fan of the test_xyzflow_exception - the goal is to test all situations, some of which result in exceptions. Test like test_xyzflow_exception imply tests should be written with a specific outcome in mind, rather than a specific situation

> I do see the appeal of ensuring every service function has proper exception handling, as you discovered some services lacked. What are your thoughts? In the generic exception scenario, the database or client likely has to throw an exception - is there an easy way to cause an erroneous test with the fixture db or client?

Reply:
I'm thinking that we can structure as:
- test_success
- test_known_failure
- test_known_failure
- ...
- test_unexpected_exception (for handling)

Need to look into it more, but I'm hoping we can use the `mock` package to throw an exception